libraries/AP_Notify/OreoLED_I2C.cpp

```diff
@@ -0,0 +1,720 @@
+        set_rgb(OREOLED_BACKRIGHT, OREOLED_PATTERN_STROBE, 255, 0, 0,0,0,0,PERIOD_SUPER,PO_ALTERNATE);
+        break;
+    }
+
+    return AP_Notify::flags.autopilot_mode;
+}
+
+
+// Procedure for when Pixhawk is in a pilot controlled mode
+// All LEDs use standard pattern and colors
+bool OreoLED_I2C::mode_pilot_flight()
+{
+    switch (_oreo_theme) {
+
+    case OreoLED_Aircraft:
+        set_rgb(OREOLED_FRONTLEFT, OREOLED_PATTERN_SOLID, 255, 0, 0);
+        set_rgb(OREOLED_FRONTRIGHT, OREOLED_PATTERN_SOLID, 0, 255, 0);
+        if ((AP_Notify::flags.pre_arm_check && AP_Notify::flags.pre_arm_gps_check) || AP_Notify::flags.armed) {
+            set_rgb(OREOLED_BACKLEFT, OREOLED_PATTERN_STROBE, _rear_color_r, _rear_color_g, _rear_color_b,0,0,0,PERIOD_FAST,0);
+            set_rgb(OREOLED_BACKRIGHT, OREOLED_PATTERN_STROBE, _rear_color_r, _rear_color_g, _rear_color_b,0,0,0,PERIOD_FAST,PO_ALTERNATE);
+        } else {
+            set_rgb(OREOLED_BACKLEFT, OREOLED_PATTERN_SOLID, _rear_color_r, _rear_color_g, _rear_color_b);
+            set_rgb(OREOLED_BACKRIGHT, OREOLED_PATTERN_SOLID, _rear_color_r, _rear_color_g, _rear_color_b);
+        }
+        break;
+
+    case OreoLED_Automobile:
+        set_rgb(OREOLED_FRONTLEFT, OREOLED_PATTERN_SOLID, 255, 255, 255);
+        set_rgb(OREOLED_FRONTRIGHT, OREOLED_PATTERN_SOLID, 255, 255, 255);
+        set_rgb(OREOLED_BACKLEFT, OREOLED_PATTERN_SOLID, 255, 0, 0);
+        set_rgb(OREOLED_BACKRIGHT, OREOLED_PATTERN_SOLID, 255, 0, 0);
+        break;
+
+    default:
+        set_rgb(OREOLED_FRONTLEFT, OREOLED_PATTERN_SOLID, 255, 255, 255);
+        set_rgb(OREOLED_FRONTRIGHT, OREOLED_PATTERN_SOLID, 255, 255, 255);
+        set_rgb(OREOLED_BACKLEFT, OREOLED_PATTERN_SOLID, 255, 0, 0);
+        set_rgb(OREOLED_BACKRIGHT, OREOLED_PATTERN_SOLID, 255, 0, 0);
+        break;
+    }
+
+    return true;
+}
+
+
+// set_rgb - Solid color settings only
+void OreoLED_I2C::set_rgb(uint8_t instance, uint8_t red, uint8_t green, uint8_t blue)
+{
+    set_rgb(instance, OREOLED_PATTERN_SOLID, red, green, blue);
+}
+
+
+// set_rgb - Set a color and selected pattern.
+void OreoLED_I2C::set_rgb(uint8_t instance, oreoled_pattern pattern, uint8_t red, uint8_t green, uint8_t blue)
+{
+    // get semaphore
+    WITH_SEMAPHORE(_sem);
+
+    // check for all instances
+    if (instance == OREOLED_INSTANCE_ALL) {
+        // store desired rgb for all LEDs
+        for (uint8_t i=0; i<OREOLED_NUM_LEDS; i++) {
+            _state_desired[i].set_rgb(pattern, red, green, blue);
+            if (!(_state_desired[i] == _state_sent[i])) {
+                _send_required = true;
+            }
+        }
+    } else if (instance < OREOLED_NUM_LEDS) {
+        // store desired rgb for one LED
+        _state_desired[instance].set_rgb(pattern, red, green, blue);
+        if (!(_state_desired[instance] == _state_sent[instance])) {
+            _send_required = true;
+        }
+    }
+
+    _send_required = true;
```

Comment:
I'm guessing this is debug/should be deleted? If not why do all the comparison above to see if anything changed when it's always going to be flagged as changed.

Reply:
yep, thanks, fixed